library/std/build.rs

```diff
@@ -1,6 +1,7 @@
 use std::env;
 
 fn main() {
+    println!("cargo::rustc-check-cfg=cfg(target_os, values(\"rtems\"))");
```

Comment:
I think this should use the lints table in `std/Cargo.toml` rather than a build.rs change (same as in library/panic_unwind/Cargo.toml). Does that work?

Could you past the full `unexpected cfg` error? I am not sure where this would be coming from since there is not currently any `cfg(target_os = "rtems")` in libc.

Reply:
Perfect that was it, thanks. I hope I have addressed most comments. Could you have a look, please. Afterwards I will squash the fixup commits to have a sane history.

Reply:
Awesome. Could you just double check the `pub mod raw` situation https://github.com/rust-lang/rust/pull/127021#discussion_r1723792832? With that fixed please do squash, the rest LGTM.